src/integer_mod_q/mat_zq/solve.rs

```diff
@@ -0,0 +1,222 @@
+// Copyright © 2023 Marvin Beckmann
+//
+// This file is part of qFALL-math.
+//
+// qFALL-math is free software: you can redistribute it and/or modify it under
+// the terms of the Mozilla Public License Version 2.0 as published by the
+// Mozilla Foundation. See <https://mozilla.org/en-US/MPL/2.0/>.
+
+//! This module contains implementations to solve linear equations over [`MatZq`]
+//! with arbitrary moduli.
+
+use super::MatZq;
+use crate::{
+    integer::Z,
+    integer_mod_q::Zq,
+    traits::{Concatenate, GetEntry, GetNumColumns, GetNumRows, Pow, SetEntry},
+};
+
+impl MatZq {
+    /// Computes a solution for a system of linear equations under a certain modulus
+    /// It solves `Ax = y` for `x` as a [MatZq].
+    /// If no solution is found, `None` is returned
+    ///
+    /// Note that this function does not compute a solution whenever there is one.
+    /// The complete correctness will be dealt with in the corresponding reference (TODO: Referenced in issue)
+    ///
+    /// Parameters:
+    /// - `y`: The syndrome for which a solution has to be computed.
+    ///
+    /// Returns a solution for the linear system or `None`, if none could be computed.
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::integer_mod_q::MatZq;
+    /// use std::str::FromStr;
+    ///
+    /// let mat = MatZq::from_str("[[2,2,3],[2,5,7]] mod 8").unwrap();
+    /// let y = MatZq::from_str("[[3],[5]] mod 8").unwrap();
+    /// let x = mat.solve_gaussian_elimination(&y).unwrap();
+    /// ```
```

Comment:
```suggestion
    /// ```
    ///
    /// Panics ...
    /// - ... if the dimensions of the matrix and syndrome mismatch
```

Reply:
The Panic messages are not helpfull by finding the mistake